utils_nlp/models/transformers/common.py

```diff
@@ -100,60 +113,118 @@ def get_default_scheduler(optimizer, warmup_steps, num_training_steps):
         )
         return scheduler
 
+    def prepare_model_and_optimizer(
```

Comment:
@saidbleik I spent quite some time thinking about the best way to do this. Let's discuss and agree on it and then change it for other classes. Hopefully this is the last time we need to change it. :)